ci/Dockerfile.circle_build

```diff
@@ -1,5 +1,6 @@
 # To build and deploy, from this directory:
 # $ docker build . -f Dockerfile.circle_build -t us.gcr.io/broad-dsp-gcr-public/workbench:buildimage-X.Y.Z .
+# Note: M1 Macs may need to add `--platform linux/x86_64/v8`
```

Comment:
Discovered as part of debugging [this issue](https://pmi-engteam.slack.com/archives/CHRN2R51N/p1701984349958949). It turns out changing this file was not necessary; see the deps.sh change below.